tests/thread_tests/local.go

```diff
@@ -14,13 +13,16 @@ import (
 func setup(t *testing.T) {
 	installChipTool(t)
 
+	otbrInstallTime := time.Now()
 	// Clean
 	utils.SnapRemove(t, otbrSnap)
 
 	// Install OTBR
 	utils.SnapInstallFromStore(t, otbrSnap, "latest/beta")
 	t.Cleanup(func() {
 		utils.SnapRemove(t, otbrSnap)
+		logs := utils.SnapLogs(t, otbrInstallTime, otbrSnap)
+		utils.WriteLogFile(t, otbrSnap, logs)
```

Comment:
In all instances, checking the logs before removal will provide a much cleaner output, before the shutdown logic kicks in.